src/app.ts

```diff
@@ -134,6 +139,51 @@ const registerMiddleware = (app: Express): void => {
     }
     next()
   })
+
+  // Access Control List
+  app.use(async (req: Request, res: Response, next: NextFunction) => {
+    const allRoutes: string[] = Object.values(Routes)
+    const guestRoutes: string[] = [
+      Routes.CONTACT,
+      Routes.FORGOT_PASSWORD,
+      Routes.HOME,
+      Routes.LOGIN,
+      Routes.REGISTER,
+      Routes.RESET_GUEST_PASSWORD,
+    ]
+    const user = req.user
+    const url = req.url
+    const method = req.method.toLowerCase()
+
+    if (!allRoutes.includes(url)) {
+      return res.status(404).send()
+    }
+
+    if (!user && guestRoutes.includes(url)) {
+      return next()
+    }
+
+    const currentUserRoles = user?.userRoles?.map((userRole) => userRole.role.name)
+
+    if (!currentUserRoles) {
+      return res.status(401).json({ message: 'This action is unauthorized.' })
+    }
```

Comment:
We can update this to allow `guest` acl:
```suggestion
     const currentUserRoles = user?.userRoles?.map((userRole) => userRole.role.name) ?? []
     const currentRoles [...currentUserRoles, 'guest'].
```
No need to check if `currentUserRoles` is empty since all activity will be given a `guest` role.

This will keep the normal logic for getting the user roles and then use the spread operator to add them into a new array that will also contain `guest`.